app/forms/capitalallowances/specialTaxSites/SpecialTaxSiteLocationFormProvider.scala

```diff
@@ -17,46 +17,66 @@
 package forms.capitalallowances.specialTaxSites
 
 import forms.PostcodeRegex
-import forms.capitalallowances.specialTaxSites.SpecialTaxSiteLocationFormProvider.{buildingName, buildingNumber, emptyBuildingDetailsError}
+import forms.capitalallowances.specialTaxSites.SpecialTaxSiteLocationFormProvider.{buildingName, buildingNumber, emptyBuildingDetailsError, postcode}
 import forms.mappings.Mappings
 import models.common.UserType
 import models.journeys.capitalallowances.specialTaxSites.SpecialTaxSiteLocation
-import play.api.data.Forms.{mapping, optional}
-import play.api.data.validation.{Constraint, Invalid, Valid, ValidationError}
+import play.api.data.Forms.mapping
 import play.api.data.{Form, FormError}
+import uk.gov.voa.play.form.Condition
+import uk.gov.voa.play.form.ConditionalMappings.mandatoryIf
 
 import javax.inject.Inject
 
 class SpecialTaxSiteLocationFormProvider @Inject() extends Mappings {
-  private val postcode                = "postcode"
-  private val maxBuildingNameLength   = 100
-  private val maxBuildingNameError    = "specialTaxSiteLocation.error.buildingName.length"
-  private val maxBuildingNumberLength = 20
-  private val maxBuildingNumberError  = "specialTaxSiteLocation.error.buildingNumber.length"
-  private val postcodeRequiredError   = (userType: UserType) => s"specialTaxSiteLocation.error.postcode.$userType"
-  private val postcodeInvalidError    = "error.postcode.invalid"
+  private val maxInputLength         = 90
+  private val maxBuildingNameError   = "specialTaxSiteLocation.error.buildingName.length" // TODO get the messages for these from Tim
+  private val maxBuildingNumberError = "specialTaxSiteLocation.error.buildingNumber.length"
+  private val postcodeRequiredError  = (userType: UserType) => s"specialTaxSiteLocation.error.postcode.$userType"
+  private val postcodeInvalidError   = "error.postcode.invalid"
 
-  private def atLeastOneRequired(userType: UserType): Constraint[SpecialTaxSiteLocation] = Constraint("constraints.atleastone") { location =>
-    if (location.buildingName.isEmpty && location.buildingNumber.isEmpty) {
-      Invalid(Seq(ValidationError(emptyBuildingDetailsError(userType))))
-    } else {
-      Valid
-    }
+//  private def atLeastOneRequired(userType: UserType): Constraint[SpecialTaxSiteLocation] = Constraint("constraints.atleastone") { location =>
+//    if (location.buildingName.isEmpty && location.buildingNumber.isEmpty) {
+//      Invalid(Seq(ValidationError(emptyBuildingDetailsError(userType))))
+//    } else {
+//      Valid
+//    }
+//  }
```

Comment:
Leaving these alternatives in commented atm to work on a better solution next PR